src/ceph-volume/ceph_volume/devices/lvm/batch.py

```diff
@@ -586,13 +599,25 @@ def get_physical_osds(devices: List[device.Device], args: argparse.Namespace) ->
     data_slots = args.osds_per_device
     if args.data_slots:
         data_slots = max(args.data_slots, args.osds_per_device)
-    rel_data_size = args.data_allocate_fraction / data_slots
-    mlogger.debug('relative data size: {}'.format(rel_data_size))
+    #rel_data_size = args.data_allocate_fraction / data_slots
+    #mlogger.debug('relative data size: {}'.format(rel_data_size))
     ret = []
     for dev in devices:
+        rel_data_size = args.data_allocate_fraction / data_slots
         if dev.available_lvm:
+            total_dev_size = dev.vg_size[0]
             dev_size = dev.vg_size[0]
+
+            if args.block_db_size is not None and not args.db_devices:
+                all_db_space = args.block_db_size * data_slots
+                dev_size -= all_db_space.b.as_int()
+
             abs_size = disk.Size(b=int(dev_size * rel_data_size))
+            mlogger.error(f'{dev_size} {abs_size} {rel_data_size}')
+
+            if not args.db_devices and args.block_db_size is not None:
```

Comment:
I see this check (technically this one swaps the order of `not args.db_devices` and `args.block_db_size is not None` compared to the other two, but functionally the same) being done a few times here. Maybe it would be worth it to store the bool this resolves to in some variable with a name implying what situation this represents. It could improve readability and make some of these lines short (for whatever that's worth)

Reply:
true, good idea 👍